motor/discovery/aws/ssm_instances.go

```diff
@@ -202,9 +231,19 @@ func ssmInstanceToAsset(account string, region string, instance types.InstanceIn
 		}
 	}
 	// add AWS metadata labels
-	asset.Labels = addAWSMetadataLabels(asset.Labels, ssmInstanceToBasicInstanceInfo(instance, region, account))
+	asset.Labels = addAWSMetadataLabels(asset.Labels, ssmInstanceToBasicInstanceInfo(instanceInfo))
 	if label, ok := asset.Labels[ImportedFromAWSTagKeyPrefix+AWSNameLabel]; ok {
 		asset.Name = label
 	}
 	return asset
 }
+
+func getProbableUsernameFromSSMPlatformName(name string) string {
```

Comment:
Nit: Can this be merged with the ec2-specific function? 
I don't think the name would differ if SSM agent is enabled

Reply:
it's more that we're looking at different things

for the ec2 instances we look at the image name
for ssm it gives us a platform name